src/main/scala/ai/privado/utility/SQLParser.scala

```diff
@@ -1,65 +1,130 @@
 
+object SqlCleaner {
+  def clean(sql: String): String = {
+    var cleanedSql = removeComments(sql)
+    cleanedSql = removeDynamicVariables(cleanedSql)
+    cleanedSql
+  }
+
+  private def removeComments(sql: String): String = {
+    // Replace /* ... */ style comments
+    var cleanedSql = sql.replaceAll("/\\*.*?\\*/", "")
+
+    // Replace -- style comments
+    cleanedSql = cleanedSql.replaceAll("--.*", "")
+
+    cleanedSql
+  }
+
+  private def removeDynamicVariables(sql: String): String = {
+    // Replace :variable style dynamic variables
+    val variablePattern = new Regex(":[a-zA-Z0-9_]+")
+    variablePattern.replaceAllIn(sql, "")
+  }
+}
+
 object SQLParser {
 
   val logger: Logger = LoggerFactory.getLogger(getClass)
 
-  def parseSqlQuery(sqlQuery: String): Option[(String, String, List[String])] = {
+  def parseSqlQuery(sqlQuery: String): Option[List[(String, String, List[String])]] = {
     try {
-      val statement: Statement = CCJSqlParserUtil.parse(new StringReader(sqlQuery))
+      val cleanedSql           = SqlCleaner.clean(sqlQuery)
+      val statement: Statement = CCJSqlParserUtil.parse(new StringReader(cleanedSql))
       statement match {
         case selectStmt: Select =>
-          val selectBody: PlainSelect = selectStmt.getSelectBody.asInstanceOf[PlainSelect]
-          val tableName: String       = selectBody.getFromItem.toString
-          val columns: List[String] =
-            selectBody.getSelectItems.flatMap { case item: SelectItem =>
-              item.toString match {
-                case f: String if f.contains("(") =>
-                  val function = CCJSqlParserUtil.parseExpression(f).asInstanceOf[Function]
-                  function.getParameters.getExpressions.map(_.toString)
-                case _ => List(item.toString)
+          selectStmt.getSelectBody match {
+            case plainSelect: PlainSelect =>
+              val tableName: String = plainSelect.getFromItem.toString
+              val columns: List[String] =
+                plainSelect.getSelectItems.flatMap { case item: SelectItem =>
+                  item.toString match {
+                    case f: String if f.contains("(") =>
+                      val function = CCJSqlParserUtil.parseExpression(f).asInstanceOf[Function]
+                      function.getParameters.getExpressions.toString.map(_.toString)
+                    case _ => List(item.toString)
+                  }
+                }.toList
+              Some(List(("SELECT", tableName, columns)))
+
+            case setOpList: SetOperationList =>
+              val selectStmts = setOpList.getSelects.toList
+              val tableNameColumnListMap = selectStmts.map { stmt =>
+                val plainSelect = stmt.asInstanceOf[PlainSelect]
+                val tableName   = plainSelect.getFromItem.toString
+                (
+                  tableName,
+                  plainSelect.getSelectItems.flatMap { case item: SelectItem =>
```

Comment:
The code block from LNo 74 to LNo 79 is repeated above as well, Let's have a function for this